### ISSUE TITLE
Villager trades still not opening up?

### ISSUE THREAD

**Issue by Galindorf** [WildBamaBoy/minecraft-comes-alive#687]
Posting again for the 1.9 version. 
Tried with an old save, a 1.9 save, a freshly generated world, and villagers spawned from eggs. 

Am I doing something wrong? I get some experience and then...nothing!

**Comment by WildBamaBoy** [comment]
Try on multiple types of villagers (Farmer, Smith, etc.). Does the issue happen on all of them?

**Comment by Galindorf** [comment]
OK, so the trading thing is more odd than I realized. 
My mod setup:
![screen shot 2016-04-23 at 11 19 28 am](https://cloud.githubusercontent.com/assets/17105217/14762657/4ebabc04-0945-11e6-8e43-21c238c62cbb.png)
Generated fresh world in creative mode to test trading. 
First thing I noticed, the trades are not synchronous with job listing. 
**Nico the Armorer**
1: 20 wool --> 1 emerald
2: 4 emerald --> 1 Shears
Trades Completed:
wool for emerald (in creative mode) = no XP given, no slots opened
So I switched to Survival Mode just in case: 
Trades Completed:
wool for emerald = XP GIVEN, no slots opened
emerald for shears = no XP given, no slots opened.
__________________
**Avril the Miner**
20 string --> 1 emerald
1 emerald--> 9 arrows
Completed both trades, no XP given, no slots opened
_______________
Nicki the Butcher
22 Wheat --> 1 emerald
19 Potato --> 1 emerald
15 Carrots --> 1 emerald
1 emerald --> 4 bread
Traded Emerald for Bread, no XP, no slots opened
_____________
**Judith the Weaponsmith**
15 string --> 1 emerald
24 coal --> 1 emerald
6 fish+1 emerald --> 6 cooked fish
Traded 6 fish, no XP no slots opened, String, same story

_____________
**Luci the Shepherd**
18 string --> 1 emerald
22 coal --> 1 emerald
6 fish+1 emerald --> 6 cooked fish
Traded 6 fish, no XP no slots opened

_____________________
**Ryder the Librarian**
15 string --> 1 emerald
1 emerald --> 8 arrows
Traded emerald for arrows, no XP, no slots opened

Weird bug which may or may not be MCA related, and is repeatable: 
If you run out of emeralds in a trade, it shows ZERO in red, and although there is no icon for the traded item to the right, you can still click on it to trade one more time. You see here I have 17 arrows and the red ZERO emeralds:
![screen shot 2016-04-23 at 11 16 52 am](https://cloud.githubusercontent.com/assets/17105217/14762718/5022d980-0947-11e6-97da-bdbbec7a4144.png)
and here I magically have 25 arrows! 
![screen shot 2016-04-23 at 11 17 05 am](https://cloud.githubusercontent.com/assets/17105217/14762723/73635c8a-0947-11e6-9b0f-64ee9aceedcf.png)

I am sort of at a loss as to why I got XP from one trade, but not others. I am attempting to recreate that.
~G

**Comment by The-Euphoria** [comment]
Yep, I'm having the exact same issue. 

All villagers have the same trades. There seems to be only 2 maybe 3 variants, the first set of trades start "Wool = Emerald" followed by "Emerald = Shears". The second set of trades starts "Wheat = Emeralds" followed by "Potato = Emerald". These are repeated on every single villager.

I even started half a dozen new worlds to be sure, and spawned in around 60-70 new villagers but they still all had the same identical trades.

**Comment by Galindorf** [comment]
This might be helpful: 

![villagertradechart](https://cloud.githubusercontent.com/assets/17105217/14790122/7e62244c-0ad5-11e6-9ee8-9a90744314d9.png)
(shamelessly ripped off from the [Minecraft Wiki](http://minecraft.gamepedia.com/Trading) )
The trades will tell you what type of villager you are *actually* interacting with by game mechanics, rather than their given titles.

**Comment by Galindorf** [comment]
Further Experimentation:
**Spawned Vanilla Village in 1.9 Vanilla snapshot Profile:**
Testificate Villagers of all types present with varying trades
in Trading, XP given & Trades level up

**Re-opened Village in Forge 1.9 profile with MCA**
Villagers have different titles, but all seem to have "brown robe" trades
(fisherman, shepherd, farmer, fletcher)
**
Re-opened Village in vanilla 1.9 snapshot Profile:**
Villagers disappeared. Alas!
________
In conclusion, it seems all villagers have the "brown robe" trades, regardless of job title.

**Comment by 4rz0** [comment]
I can confirm this.

**Comment by Galindorf** [comment]
Me again!
I opened up a previous 1.8.9 world with villagers I had already traded with, and they still had their original trades (armorer, cleric, etc). I still was not able to open new trade slots by trading with them, however. 

Also: I loaded up the newest version of Forge and the same issues persisted. I DID have some progress on the weird XP thing. In 1.9 on a fresh world I found that I did get XP from a trade, but only for the FIRST trade slot, and only after the second click of trading. It's like the first trade "didn't count" or something. Subsequent trading on second or third slots resulted in no XP. (also no trade slots opening, and everyone had brown robe trades)

**Comment by Awkanic** [comment]
Pull request to fix this added. If someone wants this for 1.8.9, the fix will need to be backported from 1.9.

**Comment by Awkanic** [comment]
It's fixed when running unobfuscated but the new release can't find the method names. I'll take another look soon.

**Comment by Awkanic** [comment]
So this issue has evolved into three separate problems, two of which are related and one which isn't. Visit These problems are:
1. Villager trades don't level up
2. Villager trades don't always provide XP
3. Villagers only have trades from profession id 0 (brown coats)

Issues 1 and 2 are due to an exception which we've identified and corrected. I'll submit a new pull request tonight with this fix which should work despite obfuscation.

Issue 3 is not solved. I am looking at it though.

**Comment by WildBamaBoy** [comment]
Fixed in @Awkanic's pull: https://github.com/WildBamaBoy/minecraft-comes-alive/pull/699https://github.com/WildBamaBoy/minecraft-comes-alive/pull/699

Update incoming

**Comment by Galindorf** [comment]
This is the best news!
